packages/graph/hash_graph/postgres_migrations/V4__ontology_functions.sql

```diff
@@ -45,60 +31,54 @@ OR REPLACE FUNCTION update_ontology_id (
   "updated_by_id" UUID
 ) RETURNS TABLE (version_id UUID, owned_by_id UUID) AS $update_ontology_id$
 DECLARE
-  _version_id UUID;
+  "_version_id" UUID;
 BEGIN
-  INSERT INTO version_ids (
-    version_id
-  ) VALUES (
-    gen_random_uuid()
-  ) RETURNING version_ids.version_id INTO _version_id;
-
-  SET CONSTRAINTS type_ids_pkey DEFERRED;
-  SET CONSTRAINTS type_id_unique DEFERRED;
   SET CONSTRAINTS type_ids_overlapping DEFERRED;
 
+  _version_id := gen_random_uuid();
+
   RETURN QUERY
   UPDATE type_ids
   SET
-    "transaction_time" = tstzrange(now(), NULL, '[)'),
-    "version" = update_ontology_id.version,
     "version_id" = _version_id,
-    "updated_by_id" = update_ontology_id.updated_by_id
+    "version" = update_ontology_id.version,
+    "updated_by_id" = update_ontology_id.updated_by_id,
+    "transaction_time" = tstzrange(now(), NULL, '[)')
   WHERE type_ids.base_uri = update_ontology_id.base_uri
     AND type_ids.transaction_time @> now()
-  RETURNING type_ids.version_id, type_ids.owned_by_id;
+  RETURNING _version_id, type_ids.owned_by_id;
 
-  SET CONSTRAINTS type_ids_pkey IMMEDIATE;
-  SET CONSTRAINTS type_id_unique IMMEDIATE;
   SET CONSTRAINTS type_ids_overlapping IMMEDIATE;
 END $update_ontology_id$ VOLATILE LANGUAGE plpgsql;
 
 CREATE
-OR REPLACE FUNCTION "update_type_ids_trigger" () RETURNS TRIGGER AS $pga$
-    BEGIN
-      IF (OLD.version != NEW.version - 1) THEN
-        RAISE EXCEPTION 'Not updating the latest id: % -> %', OLD.version, NEW.version
-        USING ERRCODE = 'invalid_parameter_value';
-      END IF;
+OR REPLACE FUNCTION "update_type_ids_trigger" () RETURNS TRIGGER AS $update_type_ids_trigger$
+BEGIN
+  IF (OLD.version != NEW.version - 1) THEN
```

Comment:
(just to preface, I haven't though through this too much)
Would we ever be in a situation where let's say:
```
OLD = 2
NEW = 3

LATEST = 4
```
would get to this point and insert a new row thinking it's a valid insertion? 

I suppose we would run into the unique check on `UNIQUE ("base_uri", "version")`, but was that the intention of the unique constraint to do specifically? I ask because I can imagine us slightly changing things leading to the above situation passing the insertion checks

Reply:
For your example this is true, this would violate the `UNIQUE` constraint. However, if `LATEST` is `4` and you try to update from `5 -> 6`, it wouldn't fail.

Reply:
Do we see this as a potential problem as we'd be skipping 5?

Reply:
Currently, we enforce that behavior in the Rest API (by passing `typeToUpdate` and adding `+1` to the version). we can remove that check here though.

As a side note: This PR only removed the leading whitespace at this location.